packages/client/lib/miner/miner.ts

```diff
@@ -0,0 +1,206 @@
+import { BN } from 'ethereumjs-util'
+import { ConsensusType } from '@ethereumjs/common'
+import { Event } from '../types'
+import { Config } from '../config'
+import { FullSynchronizer } from '../sync'
+
+export interface MinerOptions {
+  /* Config */
+  config: Config
+
+  /* FullSynchronizer */
+  synchronizer: FullSynchronizer
+}
+
+/**
+ * @module miner
+ */
+
+/**
+ * Implements Ethereum block creation and mining.
+ * @memberof module:miner
+ */
+export class Miner {
+  private config: Config
+  private synchronizer: FullSynchronizer
+  private assembling: boolean
+  private period: number
+  public running: boolean
+
+  /* global NodeJS */
+  private _nextAssemblyTimeoutId: NodeJS.Timeout | undefined
+
+  /**
+   * Create miner
+   * @param options constructor parameters
+   */
+  constructor(options: MinerOptions) {
+    this.config = options.config
+    this.synchronizer = options.synchronizer
+    this.running = false
+    this.assembling = false
+    this.period = 15000 // default: 15s period defined in ms
```

Comment:
Update: ah, ok, I've discovered the code from `start()`. Maybe rather directly put this into the constructor for visibility?

Reply:
ah good idea! not sure why I decided to set it later in `start()`, it's not like it can change during init.

I will update to this:

```suggestion
    this.period = (this.config.chainCommon.consensusConfig().period ?? 15) * 1000 // default: 15s period defined in ms
```